From 6ffe0cfc4f7d2cc0ff9f1d063bba94f3a2e06948 Mon Sep 17 00:00:00 2001
From: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>
Date: Fri, 3 Jan 2020 05:13:20 +0530
Subject: [PATCH] FROMGIT: UBUNTU: SAUCE: ASoC: Intel: ApolloLake support for SOF

BugLink: https://bugs.launchpad.net/bugs/1826181

Add ApolloLake support for SOF on legacy devices. This configuration
requires special BIOS/Coreboot or the SOF firmware signed with the
relevant authentication key, so should probably not be used by distribution.

Signed-off-by: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>
(cherry picked from commit ce4807fb24387bfedd46eef03c2859a5d4375269
git://github.com/thesofproject/linux.git)
Signed-off-by: Hui Wang <hui.wang@canonical.com>
Signed-off-by: Timo Aaltonen <timo.aaltonen@canonical.com>
---
 sound/soc/intel/boards/Kconfig | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sound/soc/intel/boards/Kconfig b/sound/soc/intel/boards/Kconfig
index 815b7c1b688c..28fb20c88c90 100644
--- a/sound/soc/intel/boards/Kconfig
+++ b/sound/soc/intel/boards/Kconfig
@@ -241,7 +241,7 @@ config SND_SOC_INTEL_DA7219_MAX98357A_GENERIC
 	select SND_SOC_DMIC
 	select SND_SOC_HDAC_HDMI
 
-if SND_SOC_INTEL_APL
+if SND_SOC_INTEL_APL || (SND_SOC_SOF_APOLLOLAKE && SND_SOC_SOF_HDA_LINK)
 
 config SND_SOC_INTEL_BXT_DA7219_MAX98357A_MACH
 	tristate "Broxton with DA7219 and MAX98357A in I2S Mode"
@@ -267,7 +267,7 @@ config SND_SOC_INTEL_BXT_RT298_MACH
 	   Say Y or m if you have such a device. This is a recommended option.
 	   If unsure select "N".
 
-endif ## SND_SOC_INTEL_APL
+endif ## SND_SOC_INTEL_APL || (SND_SOC_SOF_APOLLOLAKE && SND_SOC_SOF_HDA_LINK)
 
 config SND_SOC_INTEL_CNL_MAX98373_MACH
 	tristate "ASoC Audio driver for CNL with MAX98373 in I2S Mode"
-- 
2.17.1

